src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1568,6 +1574,18 @@ public override BoundNode VisitCall(BoundCall node)
                 ReplayReadsAndWrites(localFunc, node.Syntax, writes: true);
             }
 
+            if (MethodEnsuresTrueWhenExits(method) && arguments.Length > 0)
+            {
+                VisitTrueWhenExits(condition: arguments[0]);
```

Comment:
The beginning of `VisitCall` has a check for `method.CallsAreOmitted` with a comment to skip the visitor in the future when the `[Conditional]` attribute is not satisfied. But for methods such as `Debug.Assert` (and similar methods), it seems we should visit the condition argument regardless of whether we visit other arguments. Consider having one test with both `TestOptions.DebugDll` and `TestOptions.ReleaseDll` to ensure that.

Reply:
Added test `EnsuresNotNull_ConditionalMethodInReleaseMode`

---
In reply to: [186272483](https://github.com/dotnet/roslyn/pull/26656#discussion_r186272483) [](ancestors = 186272483)